### PR TITLE
Handle fluent getters for primitive return types

### PR DESCRIPTION
**Pull request by Machine-Maker**
Checks if the method has a primitive return type, and if the instructions with opcodes in the method follow a specific order and type so as to indicate it is a simple getter.

### CODE REVIEW

File: src/main/java/io/papermc/codebook/lvt/LvtAssignmentSuggester.java [hunk 1]
```diff
@@ -478,4 +480,35 @@ private static boolean isStringAllUppercase(final String input) {
 
         return null;
     }
+
+    // 3 instructions, load "this" local var, getfield, return - TODO maybe if there is a CAST,
+    private static final IntPredicate[] OPCODES_IN_ORDER = new IntPredicate[] {
+        i -> i == Opcodes.ALOAD, i -> i == Opcodes.GETFIELD, i -> i >= Opcodes.IRETURN && i <= Opcodes.RETURN
```
Comment (DenWav) [inline src/main/java/io/papermc/codebook/lvt/LvtAssignmentSuggester.java:486]: `ALOAD` should always be of slot `0` for `this`.

File: src/main/java/io/papermc/codebook/lvt/LvtAssignmentSuggester.java [hunk 2]
```diff
@@ -478,4 +480,35 @@ private static boolean isStringAllUppercase(final String input) {
...
+    };
+
+    private static @Nullable String suggestNameFromFluentGetter(
+            final AsmClassData owner, final AsmMethodData method, final MethodInsnNode insn) {
+        // I think it's best to only work with primitive types here, as other types should already have names
+        // and this dramatically cuts down on the number of methods analyzed because we aren't filtering by
+        // method name
+        if (!(method.descriptor().getReturnType() instanceof PrimitiveType)) {
```
Comment (DenWav) [inline src/main/java/io/papermc/codebook/lvt/LvtAssignmentSuggester.java:494]: I'd also not apply this to static methods.

**Review by DenWav — COMMENTED** [review]
Not sure what all you have targeted for this, so if there are static use-cases then let me know.

File: src/main/java/io/papermc/codebook/lvt/LvtAssignmentSuggester.java (hunk 2 above)
Comment (Machine-Maker) [reply src/main/java/io/papermc/codebook/lvt/LvtAssignmentSuggester.java]: Yeah, I logged all the static methods it was applying to, and only 1 was important, `System#currentTimeMillis` so I just singled that one out. Also I added a check to make sure the method didn't have any parameters.

File: src/main/java/io/papermc/codebook/lvt/LvtAssignmentSuggester.java (hunk 1 above)
Comment (Machine-Maker) [reply src/main/java/io/papermc/codebook/lvt/LvtAssignmentSuggester.java:486]: Do you mean that I should be checking more than just the opcode in these predicates? that I should also check the slot of the var insn here?

**Comment by Machine-Maker** [comment]
Updated for modular suggester system

**Comment by MiniDigger** [comment]
needs to skip intValue, booleanValue...., plus get

maybe we can even ignore the autoboxing and use the "inner" method? or we special case block state properties

another case for block state properties: `boolean booleanValue = hookState.getOptionalValue(ATTACHED).orElse(false);`

**Pull request merged**